crates/sail-sql/src/parse.rs

```diff
@@ -95,8 +97,99 @@ pub fn parse_file_format(parser: &mut Parser) -> SqlResult<String> {
     let token = parser.next_token();
     match &token.token {
         Token::Word(w) => Ok(w.value.to_uppercase()),
+        Token::SingleQuotedString(s) => Ok(s.to_uppercase()),
+        Token::DoubleQuotedString(s) => Ok(s.to_uppercase()),
         _ => Err(SqlError::invalid(format!(
             "Expected file format as one of ARROW, PARQUET, AVRO, CSV, etc, found: {token}"
         ))),
     }
 }
+
+pub fn parse_normalized_object_name(parser: &mut Parser) -> SqlResult<spec::ObjectName> {
+    let mut idents: Vec<String> = vec![];
+    loop {
+        idents.push(String::from(parse_normalized_identifier(parser)?));
+        if !parser.consume_token(&Token::Period) {
+            break;
+        }
+    }
+    if idents.iter().any(|ident| ident.contains('.')) {
+        idents = idents
+            .into_iter()
+            .flat_map(|ident| ident.split('.').map(String::from).collect::<Vec<String>>())
+            .collect()
+    }
+    Ok(spec::ObjectName::from(idents))
+}
+
+pub fn parse_normalized_identifier(parser: &mut Parser) -> SqlResult<spec::Identifier> {
+    let next_token = parser.next_token();
+    let ast_ident = match next_token.token {
```

Comment:
Should we support identifiers such as `foo.1m`?

A few unit tests may be useful for this function.